### PR TITLE
Test forward

### PR DESCRIPTION
**Pull request by mschauer**
I test that forward guiding in the case where the true drift is linear corresponds to the discrete transitions in Mitosis (not quite working if the true drift is linear, but the linear approximation is different.)

### CODE REVIEW

File: test/outest.jl [hunk 1]
```diff
@@ -1,46 +1,53 @@
...
 
 # Samples
-K = 400
+K = 800
 
 # define SDE function
-f(u,p,t) = p[1]*u + p[2]
-g(u,p,t) = p[3]
+f(u,p,t) = p[1]*u .+ p[2]
+g(u,p,t) = p[3] .+ 0u # needs to preserve type of u for forwardguided
```
Comment (mschauer) [inline test/outest.jl:16]: Perhaps resolved with the broadcasting changes?

File: test/outest.jl [hunk 2]
```diff
@@ -1,46 +1,53 @@
...
 
 # time span
 tstart = 0.0
 tend = 1.0
 dt = 0.02
 
-# initial condition
+# intial condition
```
Comment (mschauer) [inline test/outest.jl:23]:
```suggestion
# initial condition
```

File: test/outest.jl [hunk 3]
```diff
@@ -23,24 +24,30 @@ dt = 0.02
...
 # set of linear parameters Eq.~(2.2)
-plin = copy(p)
-pest = [2.0]
-sdekernel = MitosisStochasticDiffEq.SDEKernel(f,g,tstart,tend,pest,plin,p=p,dt=dt)
+plin = copy(par)
+pest = copy(par)
+sdekernel = MitosisStochasticDiffEq.SDEKernel(f,g,tstart,tend,pest,plin,p=par,dt=dt)
+
+plin2 = [-0.05, 0.07, 1.0]
+sdekernel2 = MitosisStochasticDiffEq.SDEKernel(f,g,tstart,tend,pest,plin2,p=par,dt=dt)
```
Comment (mschauer) [inline test/outest.jl:32]: Here I was unsure: I want the true drift have parameters [-0.1, 0.0, 0.9] and the "linearisation" (never mind it is already linear) [-0.05, 0.07, 1.0]. @frankschae

File: test/outest.jl [hunk 4]
```diff
@@ -49,16 +56,55 @@ p̂ = Gaussian{(:μ, :Σ)}(mean(samples), cov(samples))
...
+samples = [MitosisStochasticDiffEq.forwardguiding(sdekernel, message, (u0, 0.0), Z=nothing; save_noise=true)[1][1,end] for k in 1:K]
+
+@testset "Mitosis forward" begin
+    @test pT.μ[] ≈ mean(samples) atol=0.02
+    @test pT.Σ[] ≈ cov(samples) atol=0.02
+
+end
+
+
+# try titled forward
```
Comment (mschauer) [inline test/outest.jl:98]:
```suggestion
# try tilted forward
```

File: test/outest.jl [hunk 5]
```diff
@@ -49,16 +56,55 @@ p̂ = Gaussian{(:μ, :Σ)}(mean(samples), cov(samples))
...
+
+# Try forward
+m, p = Mitosis.backwardfilter(gkernel, WGaussian{(:F, :Γ, :c)}(y*10.0, 10.0I, 0.0))
+kᵒ = Mitosis.left′(BFFG(), gkernel, m)
+
+pT = kᵒ([u0])
+
+# initial values for ODE
+mynames = (:logscale, :μ, :Σ);
+myvalues = [0.0, y_, 0.1]; # start with observation μ=y with uncertainty Σ=10
```
Comment (mschauer) [inline test/outest.jl:81]:
```suggestion
myvalues = [0.0, y_, 0.1]; # start with observation μ=y with uncertainty Σ=0.1
```

File: test/outest.jl (hunk 3 above)
Comment (frankschae) [reply test/outest.jl:32]: The lines above specify different linear parameters (B, beta, \sigma tilde) .. So probably you'd like to pass different  estimated paramters pest1 pest2 -- these are the ones used in `forward_guiding`?

**Pull request merged**